src/ducks/connections/index.js

```diff
@@ -280,66 +382,113 @@ export const getConnectionStatus = (
 
   const triggerId = validTriggers[0]
 
-  if (triggers[triggerId].isRunning) return 'running'
-  if (triggers[triggerId].isConnected) return 'connected'
-  return 'errored'
+  return getTriggerConnectionStatus(triggers[triggerId])
+}
+
+export const getConnectionStatusForTrigger = (state, trigger) => {
+  if (!trigger) return null
+  const { konnector } = trigger.message
+  return getTriggerConnectionStatus(
+    !!state.konnectors &&
+      !!state.konnectors[konnector] &&
+      !!state.konnectors[konnector] &&
+      !!state.konnectors[konnector].triggers &&
+      state.konnectors[konnector].triggers[trigger._id]
+  )
 }
 
 export const getKonnectorConnectedAccount = (state, konnector) => {
-  return getKonnectorAccount(state[konnector.slug])
+  return getKonnectorAccount(state.konnectors[konnector.slug])
 }
 
-const getConnectionStatusFromTrigger = trigger => {
+// Map the trigger status to a status compatible with queue
+const getTriggerQueueStatus = trigger => {
   if (trigger.isRunning) return 'ongoing'
   if (trigger.hasError) return 'error'
   if (trigger.isConnected) return 'done'
   return 'pending'
 }
 
-export const getQueue = (state, registryKonnectors) => {
+const getTriggerConnectionStatus = trigger => {
+  if (trigger.isRunning) return 'running'
+  if (trigger.isConnected) return 'connected'
+  return 'errored'
+}
+
+export const getQueue = (state, registryKonnectors) =>
   // state is state.connections
-  return Object.keys(state).reduce((queuedConnections, konnectorSlug) => {
-    const triggers = state[konnectorSlug].triggers
-    if (!triggers) return queuedConnections
-    const registryKonnector = registryKonnectors.data[konnectorSlug]
-    return queuedConnections.concat(
-      Object.keys(triggers).reduce((queuedTriggers, triggerId) => {
-        if (triggers[triggerId].isEnqueued) {
-          const label = registryKonnector.name
-          const status = getConnectionStatusFromTrigger(triggers[triggerId])
-          const icon = getKonnectorIcon(registryKonnector)
-          return queuedTriggers.concat({ label, status, icon })
-        }
+  state.konnectors
+    ? Object.keys(
+        state.konnectors
+      ).reduce((queuedConnections, konnectorSlug) => {
+        const triggers = state.konnectors[konnectorSlug].triggers
+        if (!triggers) return queuedConnections
+        const registryKonnector = registryKonnectors.data[konnectorSlug]
+        return queuedConnections.concat(
+          Object.keys(triggers).reduce((queuedTriggers, triggerId) => {
+            if (triggers[triggerId].isEnqueued) {
+              const label = registryKonnector.name
+              const status = getTriggerQueueStatus(triggers[triggerId])
+              const icon = getKonnectorIcon(registryKonnector)
+              return queuedTriggers.concat({ label, status, icon })
+            }
 
-        return queuedTriggers
+            return queuedTriggers
+          }, [])
+        )
```

Comment:
i've read the above block a couple of times and can't understand it, there's just too much happening. By the time I read the last `reduce` code I have no idea what i'm reducing and why. I think it would be good to split this code into more lines, with more named variables.